src/devices/Ft4222/Ft4222I2c.cs

```diff
@@ -0,0 +1,145 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Device.Ft4222;
+using System.IO;
+
+namespace System.Device.I2c
+{
+    /// <summary>
+    /// FT4222 I2C Device
+    /// </summary>
+    public class Ft4222I2c : I2cDevice
+    {
+        private I2cConnectionSettings _settings;
+        private IntPtr _ftHandle = new IntPtr();
+
+        /// <summary>
+        /// Store the FTDI Device Information
+        /// </summary>
+        public DeviceInformation DeviceInformation { get; internal set; }
+
+        /// <summary>
+        /// Create a FT4222 I2C Device
+        /// </summary>
+        /// <param name="settings">I2C Connection Settings</param>
+        public Ft4222I2c(I2cConnectionSettings settings)
+        {
+            _settings = settings;
+            // Check device
+            var devInfos = FtCommon.GetDevices();
+            // Select the one from bus Id
+            DeviceInformation = devInfos[_settings.BusId];            
+            
+            // Open device
+            var ftStatus = FtFunction.FT_OpenEx(DeviceInformation.LocId, FtOpenType.OpenByLocation, ref _ftHandle);
+
+            if (ftStatus != FtStatus.Ok)
+                throw new IOException($"Failed to open device {DeviceInformation.Description}, status: {ftStatus}");
+
+            // Set the clock
+            FtClockRate ft4222Clock = FtClockRate.Clock24MHz;
+
+            ftStatus = FtFunction.FT4222_SetClock(_ftHandle, ft4222Clock);
+            if (ftStatus != FtStatus.Ok)
+                throw new IOException($"Failed set clock rate {ft4222Clock} on device: {DeviceInformation.Description}, status: {ftStatus}");
+
+            // Set the device as I2C Master
+            ftStatus = FtFunction.FT4222_I2CMaster_Init(_ftHandle, 400);
+            if (ftStatus != FtStatus.Ok)
+                throw new IOException($"Failed to initialize I2C Master mode on device: {DeviceInformation.Description}, status: {ftStatus}");
+        }
+
+        /// <summary>
+        /// The connection settings of a device on an I2C bus. The connection settings are immutable after the device is created
+        /// so the object returned will be a clone of the settings object.
+        /// </summary>
+        public override I2cConnectionSettings ConnectionSettings => _settings;
+
+        /// <summary>
+        /// Reads data from the I2C device.
+        /// </summary>
+        /// <param name="buffer">
+        /// The buffer to read the data from the I2C device.
+        /// The length of the buffer determines how much data to read from the I2C device.
+        /// </param>
+        public override void Read(Span<byte> buffer)
+        {
+            byte[] buff = new byte[buffer.Length];
+            ushort byteRead = 0;
+            var ftStatus = FtFunction.FT4222_I2CMaster_Read(_ftHandle, (ushort)_settings.DeviceAddress, buff, (ushort)buff.Length, ref byteRead);
```

Comment:
can't span be passed directly here? You can convert it to pointer with fixed statement or use `ref byte` in the signature  and pass in `ref MemoryMarshal.GetReference(buffer)`

Reply:
I confirm, it's working. I'll adjust all including the SPI functions as well.